File: src/main/java/land/leets/global/error/ErrorResponse.java
package land.leets.global.error;

import lombok.Getter;

@Getter
public class ErrorResponse {
    
    private final int httpStatus;
    private final String message;
    private final String code;

    public ErrorResponse(ErrorCode errorCode) {
        this.httpStatus = errorCode.getHttpStatus();
        this.message = errorCode.getMessage();
        this.code = errorCode.getCode();
    }

    public ErrorResponse(ErrorCode errorCode, String customMessage) {
        this.httpStatus = errorCode.getHttpStatus();
        this.message = customMessage;
        this.code = errorCode.getCode();
    }

    public static ErrorResponse of(ErrorCode errorCode) {
        return new ErrorResponse(errorCode);
    }
}

File: src/main/kotlin/land/leets/global/error/ErrorResponse.kt
package land.leets.global.error

data class ErrorResponse(
    val httpStatus: Int,
    val message: String,
    val code: String
) {

    companion object {
        fun from(errorCode: ErrorCode) =
            ErrorResponse(
                errorCode.httpStatus,
                errorCode.message,
                errorCode.code
            )

        fun of(
            errorCode: ErrorCode,
            customMessage: String
        ) = ErrorResponse(
            errorCode.httpStatus,
            customMessage,
            errorCode.code
        )
    }
}
